File: 3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.cpp
class Solution {
public:
    bool checkDivisibility(int n) {
        int original = n;
        int summ = 0;
        int prod = 1;
        while (n > 0){
            int last_digit = n % 10;
            summ += last_digit;
            prod *= last_digit;
            n /= 10;
        }
        return original % (summ + prod) == 0;
    }
};

File: 3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        summ = 0
        prod = 1
        temp = n
        while n > 0 :
            last_digit = n % 10
            summ += last_digit
            prod *= last_digit
            n //= 10
        total = summ + prod
        if temp % total == 0:
            return True
        else:
            return False